### PR TITLE
feat: Add support for web push notifications

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server-push-adapter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server-push-adapter/issues?q=is%3Aissue).

### Issue Description

I've added Parse.Installation support https://github.com/parse-community/Parse-SDK-JS/pull/2119 to the JS SDK. The user can now save a `deviceToken` and `deviceType` however their is no way to use `Parse.Push` with this.

### Approach

I've tested this in production and it works great!

Client
```
const register = await navigator.serviceWorker.register('./worker.js', { scope: '/' });
const subscription = await register.pushManager.subscribe({
  userVisibleOnly: true,
  applicationServerKey: publicVapidKey,
});
const installation = await Parse.CoreManager.getInstallationController().currentInstallation();
installation.set('deviceType', 'web');
installation.set('deviceToken', JSON.stringify(subscription));
await installation.save();
```
Service Worker (worker.js)
```
self.addEventListener('push', (e) => {
  const data = e.data.json();
  self.registration.showNotification(data.alert, {});
});
```
Server
```
const PushAdapter = require('@parse/push-adapter').default;
const parseServerOptions = {
  push: {
    adapter: new PushAdapter({
      web: {
        /*  [options](https://github.com/web-push-libs/web-push#api-reference) */
        vapidDetails: {
           subject: 'test@example.com',
           publicKey: 'publicVapidKey',
           privateKey: 'privateKey',
        },
      },
    })
  },
  /* Other Parse Server options */
};

const query = new Parse.Query(Parse.Installation);
query.equalTo('deviceType', 'web');

await Parse.Push.send({
  where: query,
  data: {
    alert: 'Hello World!',
  },
}, { useMasterKey: true });
```

### TODOs before merging

- [x] Add tests
- [x] Add entry to changelog
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

File: spec/ParsePushAdapter.spec.js [hunk 1]
```diff
@@ -22,12 +23,20 @@ describe('ParsePushAdapter', () => {
...
+    expect(typeof ParsePushAdapterPackage.WEB).toBe('function');
     expect(typeof ParsePushAdapterPackage.utils).toBe('object');
   });
 
   it('can be initialized', (done) => {
     // Make mock config
     var pushConfig = {
+      web: {
+        vapidDetails: {
+          subject: 'test@test.com',
```
Comment (mtrezza) [inline spec/ParsePushAdapter.spec.js:35]:
```suggestion
          subject: 'test@example.com',
```

File: spec/ParsePushAdapter.spec.js [hunk 2]
```diff
@@ -337,6 +369,13 @@ describe('ParsePushAdapter', () => {
 
   it('reports properly results', (done) => {
     var pushConfig = {
+      web: {
+        vapidDetails: {
+          subject: 'test@test.com',
```
Comment (mtrezza) [inline spec/ParsePushAdapter.spec.js:374]:
```suggestion
          subject: 'test@example.com',
```

File: spec/WEB.spec.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+const pushSubscription = {
+  endpoint: '',
+  keys: {
+    p256dh: '',
+    auth: '',
+  },
+};
+
+const vapidDetails = {
+  subject: 'test@test.com',
```
Comment (mtrezza) [inline spec/WEB.spec.js:13]:
```suggestion
  subject: 'test@example.com',
```

**Review by mtrezza — APPROVED** [review]
Looks good, waiting for CI to pass...

**Comment by dplewis** [comment]
@mtrezza Codecov is broken

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0](https://github.com/parse-community/parse-server-push-adapter/releases/tag/6.1.0)

**Pull request merged**